coretex/networking/network_manager_base.py

```diff
@@ -300,6 +302,9 @@ def sampleDownload(
             print(">> [Coretex] Retry count: {0}".format(retryCount))
             return self.sampleDownload(endpoint, destination, ignoreCache, parameters, retryCount + 1)
 
+        if Path(destination).exists():
+            os.utime(destination, (os.stat(destination).st_atime, time.time()))
```

Comment:
Does it make more sense to implement this in `Sample.download` function?

Reply:
It may make more sense there